etcd/cluster.go

```diff
@@ -19,6 +19,8 @@ func NewCluster(machines []string) *Cluster {
 		machines = []string{"http://127.0.0.1:4001"}
 	}
 
+	shuffleStringSlice(machines)
```

Comment:
When do you mean when update ? If you said is on each failover, I don't think we should shuffle a new list. It will violate the round-robin.